File: clients/console-client/src/main/java/ec/edu/grupo3/client/controllers/ConversionController.java
package ec.edu.grupo3.client.controllers;

import ec.edu.grupo3.client.models.SoapServiceModel;
import ec.edu.grupo3.client.models.UserSession;
import ec.edu.grupo3.client.views.MenuView;
import java.util.Arrays;
import java.util.List;

public class ConversionController {

    private final MenuView menuView = new MenuView();
    private final SoapServiceModel model = new SoapServiceModel();
    private final UserSession session;

    private final List<String> massUnits = Arrays.asList("KILOGRAM", "GRAM", "POUND", "OUNCE");
    private final List<String> lengthUnits = Arrays.asList("METER", "KILOMETER", "CENTIMETER", "MILE", "YARD", "FOOT", "INCH");
    private final List<String> tempUnits = Arrays.asList("CELSIUS", "FAHRENHEIT", "KELVIN");

    public ConversionController(UserSession session) {
        this.session = session;
    }

    public void start() {
        while (true) {
            int choice = menuView.showMainMenu();
            switch (choice) {
                case 0:
                    processConversion("Mass", massUnits);
                    break;
                case 1:
                    processConversion("Length", lengthUnits);
                    break;
                case 2:
                    processConversion("Temperature", tempUnits);
                    break;
                case 3:
                    return;
            }
        }
    }

    private void processConversion(String category, List<String> units) {
        String from = menuView.selectUnit("SELECCIONE UNIDAD DE ORIGEN (" + category + ")", units);
        String to = menuView.selectUnit("SELECCIONE UNIDAD DE DESTINO (" + category + ")", units);
        double inputVal = menuView.getInputValue(from);

        try {
            double resultVal = 0;
            if ("Mass".equals(category)) resultVal = model.convertMass(inputVal, from, to);
            else if ("Length".equals(category)) resultVal = model.convertLength(inputVal, from, to);
            else if ("Temperature".equals(category)) resultVal = model.convertTemperature(inputVal, from, to);

            menuView.showResult(category, inputVal, from, resultVal, to);
        } catch (Exception e) {
            menuView.showError(e.getMessage() != null ? e.getMessage() : "Error de conexión al servidor Payara.");
        }
    }
}

File: clients/console-client/Controllers/ConversionController.cs
using System;
using System.Threading.Tasks;
using ConsoleClient.Models;
using ConsoleClient.Views;

namespace ConsoleClient.Controllers
{
    public class ConversionController
    {
        private readonly UserSession _session;
        private readonly MenuView _menuView = new MenuView();
        private readonly ConversionView _conversionView = new ConversionView();
        
        private readonly string[] _massUnits = { "KILOGRAM", "GRAM", "POUND", "OUNCE" };
        private readonly string[] _lengthUnits = { "METER", "KILOMETER", "CENTIMETER", "MILE", "YARD", "FOOT", "INCH" };
        private readonly string[] _tempUnits = { "CELSIUS", "FAHRENHEIT", "KELVIN" };

        public ConversionController(UserSession session)
        {
            _session = session;
        }

        public async Task RunAsync()
        {
            while (true)
            {
                int choice = _menuView.RenderMainMenu(_session.Username);

                switch (choice)
                {
                    case 0:
                        await ProcessConversionAsync("Mass", "CONVERSIÓN DE MASA", _massUnits);
                        break;
                    case 1:
                        await ProcessConversionAsync("Length", "CONVERSIÓN DE LONGITUD", _lengthUnits);
                        break;
                    case 2:
                        await ProcessConversionAsync("Temperature", "CONVERSIÓN DE TEMPERATURA", _tempUnits);
                        break;
                    case 3:
                    case -1:
                        ConsoleUIHelper.PrintHeader("CERRANDO SESIÓN");
                        Console.WriteLine("  ¡Gracias por usar el sistema de conversión SOAP!");
                        return;
                }
            }
        }

        private async Task ProcessConversionAsync(string category, string title, string[] units)
        {
            var data = _conversionView.RenderConversionPrompt(title, units);
            _conversionView.RenderLoading();

            using var invoker = new ServiceInvoker();
            try
            {
                var response = await invoker.InvokeConversionAsync(category, data.value, data.fromUnit, data.toUnit);
                _conversionView.RenderResult(response);
            }
            catch (Exception ex)
            {
                _conversionView.RenderError(ex.Message);
            }
        }
    }
}